likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedFragment.kt

```diff
@@ -0,0 +1,220 @@
+                        }
+                    }
+                }
+            setPaginationScrollListener(paginationScrollListener)
+        }
+    }
+
+    //todo: set loader color using style
+    private fun initSwipeRefreshLayout() {
+        mSwipeRefreshLayout = binding.swipeRefreshLayout
+        mSwipeRefreshLayout.setColorSchemeColors(
+            //todo: change this color as per the style
+            ContextCompat.getColor(
+                requireContext(),
+                R.color.lm_feed_majorelle_blue
+            )
+        )
+
+        mSwipeRefreshLayout.setOnRefreshListener {
+            refreshNotificationFeed()
+        }
+    }
+
+    //refresh the whole notification feed
+    private fun refreshNotificationFeed() {
+        mSwipeRefreshLayout.isRefreshing = true
+        binding.rvActivityFeed.resetScrollListenerData()
+        activityFeedViewModel.getActivityFeed(1)
+    }
+
+    private fun fetchData() {
+        LMFeedProgressBarHelper.showProgress(binding.progressBar)
+        activityFeedViewModel.getActivityFeed(1)
+    }
+
+    private fun observeData() {
+        // observes get notification feed response
+        activityFeedViewModel.activityFeedResponse.observe(viewLifecycleOwner) { pair ->
+            observeNotificationFeed(pair)
+        }
+
+        activityFeedViewModel.errorMessageEventFlow.onEach { response ->
+            observeErrorMessage(response)
+        }.observeInLifecycle(viewLifecycleOwner)
+    }
+
+    private fun observeNotificationFeed(pair: Pair<Int, List<LMFeedActivityViewData>>) {
+        //hide progress bar
+        LMFeedProgressBarHelper.hideProgress(binding.progressBar)
+
+        //page in api send
+        val page = pair.first
+
+        //list of activities
+        val activities = pair.second
+
+        //if pull to refresh is called
+        if (mSwipeRefreshLayout.isRefreshing) {
+            setFeedAndScrollToTop(activities)
+            mSwipeRefreshLayout.isRefreshing = false
+            return
+        }
+
+        //normal adding
+        if (page == 1) {
+            checkForNoActivity(activities)
+            setFeedAndScrollToTop(activities)
+        } else {
+            binding.rvActivityFeed.addActivities(activities)
+        }
+    }
+
+    //set activities through diff utils and scroll to top of the feed
+    private fun setFeedAndScrollToTop(feed: List<LMFeedActivityViewData>) {
+        binding.rvActivityFeed.apply {
+            replaceActivities(feed)
+            scrollToPosition(0)
+        }
+    }
+
+    //todo:
+    //checks if there is any activity or not
+    private fun checkForNoActivity(feed: List<LMFeedBaseViewType>) {
+        if (feed.isNotEmpty()) {
+            binding.apply {
+                //todo:
+//                layoutNoNotification.root.hide()
+                rvActivityFeed.show()
+            }
+        } else {
+            binding.apply {
+//                layoutNoNotification.root.show()
```

Comment:
why this is commented?

Reply:
added in further PRs. The no layout view was not ready till this time